plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesStageMojo.java

```diff
@@ -1027,7 +1100,15 @@ static String generateFlexTemplateImagePath(
     return Optional.ofNullable(artifactRegistry)
         .map(
             value ->
-                value + "/" + projectIdUrl + "/" + stagePrefix.toLowerCase() + "/" + containerName)
+                value.contains("gcr.io")
+                    ? value
+                        + "/"
+                        + projectIdUrl
+                        + "/"
+                        + stagePrefix.toLowerCase()
+                        + "/"
+                        + containerName
+                    : value + "/" + stagePrefix.toLowerCase() + "/" + containerName)
```

Comment:
Can we rename `generateStagingFlexTemplateImagePath` to `generateImagePath` and then use it to construct these?

Reply:
This method mirrors `generateFlexTemplateImagePath` which is used generate the final image path so a "Staging" infix is added

Reply:
My point is that this is doing the same thing that `generateStagingFlexTemplateImagePath` is doing (combining a registry, path, and containerName), so it seems like it is something we should share.

Though I guess that leads to the question, why do we need this extra logic for the final path, but not the staging path? Aren't they really doing the same thing, just with a different artifact registry instance?

Reply:
In other words, instead of calling `generateStagingFlexTemplateImagePath(containerName, stagingArtifactRegistry, stagePrefix)`, wouldn't we get an equal/slightly more robust outcome by calling `generateFlexTemplateImagePath(containerName, stagingArtifactRegistry, stagePrefix)`?

Reply:
Thanks, consolidated two methods, and ran a mvn verify again confirmed it still work